### PR TITLE
[IMP] project: generic improvement for project configuration

### PR DESCRIPTION
**Pull request by mba-odoo**
Purpose of the task is to do a generic UX improvement in project

Specification:
--------------
1) Stages(project.task.type)

* List view:
  - move the 'folded in kanban' field to the right of the project_ids one
  - remove the description field from the list view
  - make the list view editable in batch
* Form view:
  - rename 'stage name' into 'name'
  - move the sequence field below the 'rating email template' one
  - change the copy of the tooltips to: "At each stage, employees can block tasks or mark them as ready for the next step. You can customize here the labels for each state."
  - the tags should be colored according to the project's color
  - reduce the space in between the icons and the name of the states
  - add some space in between the legend_done field and the description
  - when creating a new mail template, set 'project.task' as model_id of the template
* move the 'Stages' menu into below the 'Projects' one
* add the project_ids field to the kanban view
* search: rename 'Tasks Stages' into 'Name'

2) Tags(project.tags):

- configuration > tags menu: change the helper to: "Tags are perfect for categorizing your tasks."
- add an optional list view -> only the color field should be optional and displayed by default
- new tags should be created at the top of the list

- Also removed some of the unused(deprecated) css which are not applied anywhere in the project.

TaskID: 2508642

### CODE REVIEW

File: addons/project/views/project_views.xml [hunk 1]
```diff
@@ -105,30 +106,34 @@
...
+                                    title="Task is blocked. Click to unblock or set as done."
+                                    aria-label="Task is blocked. Click to unblock or set as done." role="img"/>
+                                <field name="legend_blocked" nolabel="1"/>
+                            </div>
+                            <div class="o_row" colspan="2">
+                                <label for="legend_done" string=" " class="o_status o_status_green oe_project_kanban_legend"
+                                    title="This step is done. Click to block or set in progress."
+                                    aria-label="This step is done. Click to block or set in progress." role="img"/>
+                                <field name="legend_done" nolabel="1"/>
+                            </div>
```
Comment (xavierbol) [inline addons/project/views/project_views.xml:134]: Good job to keep the `oe_project_kanban_legend` class, however, if you look the view in mobile view, then you will see the label is not aligned with the field. 
I'm not an expert in CSS, but I think we can do the same using flexbox to keep the same rendering like this: 
```suggestion
                            <div class="row ml-1" colspan="2">
                                <label for="legend_normal" string=" " class="o_status"
                                    title="Task in progress. Click to block or set as done."
                                    aria-label="Task in progress. Click to block or set as done." role="img"/>
                                <field name="legend_normal" nolabel="1"/>
                            </div>
                            <div class="row ml-1" colspan="2">
                                <label for="legend_blocked" string=" " class="o_status o_status_red"
                                    title="Task is blocked. Click to unblock or set as done."
                                    aria-label="Task is blocked. Click to unblock or set as done." role="img"/>
                                <field name="legend_blocked" nolabel="1"/>
                            </div>
                            <div class="row ml-1" colspan="2">
                                <label for="legend_done" string=" " class="o_status o_status_green"
                                    title="This step is done. Click to block or set in progress."
                                    aria-label="This step is done. Click to block or set in progress." role="img"/>
                                <field name="legend_done" nolabel="1"/>
                            </div>
```

**Review by xavierbol — COMMENTED** [review]
Good job, I have just a small suggestion.
Thank you for removing the unnecessary CSS classes. 🙂

File: addons/project/views/project_views.xml (hunk 1 above)
Comment (mba-odoo) [reply addons/project/views/project_views.xml]: @xavierbol Done the changes with several modification. Thanks

**Review by IT-Ideas — APPROVED** [review]
LGTM Thanks for your work 🙂!

**Comment by IT-Ideas** [comment]
@robodoo r+

**Comment by IT-Ideas** [comment]
@robodoo r+

**Pull request closed without merging**